packages/react/src/components/TextField/TextField.styles.tsx

```diff
@@ -201,6 +201,12 @@ export function getStyles(props: ITextFieldStyleProps): ITextFieldStyles {
         alignItems: 'stretch',
         position: 'relative',
       },
+      !focused && {
+        ':focus-within': {
+          border: `2px solid ${palette.themePrimary}`,
```

Comment:
Quick note here even though I think the actual fix is elsewhere: in general we should be using colors from `semanticColors` not `palette`. In this case it would be `semanticColors.focusBorder`.